src/view-backend-exportable-private.cpp

```diff
@@ -128,10 +146,17 @@ gboolean ViewBackend::s_socketCallback(GSocket* socket, GIOCondition condition,
     if (len == -1)
         return FALSE;
 
-    if (len == sizeof(uint32_t) * 2 && message[0] == 0x42) {
-        auto& viewBackend = *static_cast<ViewBackend*>(data);
-        viewBackend.m_id = message[1];
-        viewBackend.m_client = WS::Instance::singleton().registerViewBackend(viewBackend.m_id, viewBackend);
+    if (len != sizeof(uint32_t) * 2)
+        return TRUE;
+
+    auto& viewBackend = *static_cast<ViewBackend*>(data);
+    switch (message[0]) {
+    case 0x42:
```

Comment:
How about we define an `enum` for the message types? Something like:

```c
enum MessageType {
    RegisterSurface = 0x42,
    UnregisterSurface = 0x43,
};
```

and then in here:

```cpp
switch (message[0]) {
    case MessageType::RegisterSurface: /* ... */
    case MessageType::UnregisterSurface: /* ... */
    default: g_assert_not_reached();
}
```

Reply:
I agree too, and I thought about it, but again, I didn't want to change too much the existing code as part of this patch. I'm going to push a follow up patch to factor out the ipc code fixing all these style issues.